server/lib/service/google_api.dart

```diff
@@ -0,0 +1,66 @@
+import 'dart:async';
+
+import 'package:googleapis_auth/auth_io.dart';
+import 'package:http/http.dart' as http;
+
+/// A class that provides authenticated clients to interact with Google APIs.
+class GoogleApi {
+  /// The scopes the service account client needs to access Cloud Scheduler
+  /// and Firestore.
+  static const List<String> serviceAccountScope = [
+    'https://www.googleapis.com/auth/cloud-platform',
+    'https://www.googleapis.com/auth/cloud-scheduler',
+    'https://www.googleapis.com/auth/datastore',
+  ];
+
+  /// The scopes the user account client needs to access the ID token and
+  /// DV360 API.
+  static const List<String> userAccountScope = [
+    'openid email',
+    'https://www.googleapis.com/auth/display-video',
+  ];
+
+  /// The client ID for OAuth2.
+  final String _clientId;
+
+  /// The client secret for OAuth2.
+  final String _clientSecret;
+
+  /// Creates an instance of [GoogleApi] with a client ID and secret.
+  GoogleApi(this._clientId, this._clientSecret);
+
+  /// Creates an [AutoRefreshingAuthClient] for the service account.
+  ///
+  /// Locally, we use the JSON file generated using the following command:
+  /// ```
+  /// gcloud auth application-default login
+  /// ```
+  /// On AppEngine, the service account is used.
+  Future<AutoRefreshingAuthClient> getServiceAccountClient() async {
```

Comment:
is this used to access firestore?

Reply:
It's used to access firestore and scheduler, yes.

Reply:
I'll add more to the documentation comment.